torch/_functorch/compile_utils.py

```diff
@@ -50,6 +51,37 @@ def fx_graph_cse(fx_g: torch.fx.graph.Graph):
     )
 
     compute_mutation_region_ids(fx_g)  # type: ignore[arg-type]
+
+    # Make a set of separate storages returned from the output, which will be preserved
+    # when pruning.  This prevents us from deduplicating returned tensors which have
+    # experienced identical operations, but are separate data structures in eager mode.
+    output_node: fx.Node = list(fx_g.nodes)[-1]
+    assert output_node.op == "output"
+
+    def checkable_node(node: fx.Node) -> bool:
+        """We can evaluate only nodes that represent tensors with defined storage."""
+        if "val" not in node.meta or not isinstance(node.meta["val"], torch.Tensor):
+            return False
+
+        try:
+            node.meta["val"].untyped_storage()
```

Comment:
nit: instead of try-catching, you can just check if the value here is an actual tensor (which is one-to-one with having a storage):
```
return isinstance(node.meta['val'], torch.Tensor)
```

Reply:
@bdhirsh That actually ended up being untrue, this call throws when a `SparseTensor` is the val here.

Reply:
ah welp. FWIW, we branch on is_sparse in a few other places when attempting to grab storage (e.g. [here](https://github.com/pytorch/pytorch/blob/main/torch/_subclasses/fake_tensor.py#L993))